### PR TITLE
Add tiup microservices section

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17465
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiup/tiup-cluster-topology-reference.md [hunk 1]
```diff
@@ -77,6 +79,8 @@ The `global` section corresponds to the cluster's global configuration and has t
 
 - `arch`: The CPU architecture of the target machine. The field controls which platform to adapt to for the binary packages pushed to the target machine. The supported values are "amd64" and "arm64". The default value is "amd64".
 
+- `pd_mode`: The PD startup mode. The field controls whether to enable [PD microservices](/pd-microservices.md). The supported value is "ms". Specifying this field means enabling PD microservices.
```
Comment (qiancai) [inline tiup/tiup-cluster-topology-reference.md:82]:
```suggestion
- `pd_mode`: The PD working mode. The field controls whether to enable [PD microservices](/pd-microservices.md). The supported value is "ms". Specifying this field means enabling PD microservices.
```

File: tiup/tiup-playground.md [hunk 2]
```diff
@@ -30,44 +30,53 @@ The command-line flags of the playground component are described as follows:
...
+      --host string                  Specify the listening address of each component (default: `127.0.0.1`). Set it to `0.0.0.0` if provided for access of other machines
+      --kv int                       Specify the number of TiKV instances (default: 1)
+      --kv.binpath string            Specify the TiKV instance binary path (optional, for debugging)
+      --kv.config string             Specify the TiKV instance configuration file (optional, for debugging)
+      --mode string                  Specify the playground mode: 'tidb' (default) and 'tikv-slim'
+      --pd int                       Specify the number of PD instances (default: 1)
+      --pd.host host                 Specify the listening address of PD
+      --pd.binpath string            Specify the PD instance binary path (optional, for debugging)
+      --pd.config string             Specify the PD instance configuration file (optional, for debugging)
+      --pd.mode string               Specify the PD startup mode. The optional value is 'ms'. Specifying this flag means enabling PD microservice mode.
```
Comment (qiancai) [inline tiup/tiup-playground.md:52]:
```suggestion
      --pd.mode string               Specify the PD working mode. The optional value is 'ms'. Specifying this flag means enabling PD microservice mode.
```

File: scale-microservices-using-tiup.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+| 10.0.1.1   | TiKV   |
+| 10.0.1.2   | TiKV   |
+| 10.0.1.6   | TSO   |
+| 10.0.1.7   | Scheduling   |
+
+## Add TSO/Scheduling nodes
+
+> **Note:**
+>
+> To add TSO/Scheduling nodes to a TiDB cluster that have not enabled PD microservices yet, follow the instructions in [Switch from regular mode to microservices mode](#Switching-from-Regular-Mode-to-Microservices-Mode) instead.
```
Comment (qiancai) [inline scale-microservices-using-tiup.md:27]:
```suggestion
> To add TSO/Scheduling nodes to a TiDB cluster that have not enabled PD microservices yet, follow the instructions in [Switch from regular mode to microservices mode](#switch-from-regular-mode-to-microservices-mode) instead.
```

File: scale-microservices-using-tiup.md [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+| 10.0.1.6   | TSO   |
+| 10.0.1.7   | Scheduling   |
+| 10.0.1.8   | TSO   |
+| 10.0.1.9   | Scheduling   |
+
+## Remove TSO/Scheduling nodes
+
+> **Note:**
+>
+> For a cluster with PD microservices enabled, if you need to switch it to non-microservice mode, follow the instructions in [Switch from microservices mode to regular mode](#Switching-from-Microservices-Mode-to-Regular-Mode) instead.
```
Comment (qiancai) [inline scale-microservices-using-tiup.md:117]:
```suggestion
> For a cluster with PD microservices enabled, if you need to switch it to non-microservice mode, follow the instructions in [Switch from microservices mode to regular mode](#switch-from-microservices-mode-to-regular-mode) instead.
```

File: command-line-flags-for-scheduling-configuration.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+# Scheduling Configuration Flags
+
+The Scheduling node is used for providing the `scheduling` microservice for PD. You can configure it using command-line flags or environment variables.
+
+## `--advertise-listen-addr`
+
+- The URL for the client to access the Scheduling node.
+- Default: `${listen-addr}`
+- In some situations such as in Docker or NAT network environments, if a client cannot access the Scheduling node through the default client URLs listened to by `scheduling`, you must manually set `--advertise-listen-addr` for client access.
+- For example, the internal IP address of Docker is `172.17.0.1`, while the IP address of the host is `192.168.100.113` and the port mapping is set to `-p 3379:3379`. In this case, you can set `--advertise-listen-addr="http://192.168.100.113:3379"`. Then, the client can find this service through `"http://192.168.100.113:3379"`.
```
Comment (Oreoxmt) [inline command-line-flags-for-scheduling-configuration.md:15]:
```suggestion
- For example, the internal IP address of Docker is `172.17.0.1`, while the IP address of the host is `192.168.100.113` and the port mapping is set to `-p 3379:3379`. In this case, you can set `--advertise-listen-addr="http://192.168.100.113:3379"`. Then, the client can find this service through `http://192.168.100.113:3379`.
```

File: command-line-flags-for-tso-configuration.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+# TSO Configuration Flags
+
+The TSO node is used for providing the `tso` microservice for PD. You can configure it using command-line flags or environment variables.
+
+## `--advertise-listen-addr`
+
+- The URL for the client to access the TSO node.
+- Default: `${listen-addr}`
+- In some situations such as in Docker or NAT network environments, if a client cannot access the TSO node through the default client URLs listened to by `tso`, you must manually set `--advertise-listen-addr` for client access.
+- For example, the internal IP address of Docker is `172.17.0.1`, while the IP address of the host is `192.168.100.113` and the port mapping is set to `-p 3379:3379`. In this case, you can set `--advertise-listen-addr="http://192.168.100.113:3379"`. Then, the client can find this service through `"http://192.168.100.113:3379"`.
```
Comment (Oreoxmt) [inline command-line-flags-for-tso-configuration.md:15]:
```suggestion
- For example, the internal IP address of Docker is `172.17.0.1`, while the IP address of the host is `192.168.100.113` and the port mapping is set to `-p 3379:3379`. In this case, you can set `--advertise-listen-addr="http://192.168.100.113:3379"`. Then, the client can find this service through `http://192.168.100.113:3379`.
```

File: pd-microservices-deployment-topology.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+</details>
+
+For detailed descriptions of the configuration items in the preceding TiDB cluster topology file, see [Topology configuration file for deploying TiDB using TiUP](/tiup/tiup-cluster-topology-reference.md).
+
+### Key parameters
+
+- The instance-level `"-host"` configuration in `tso_servers` only supports IP address, not domain name.
+- For detailed descriptions of TSO configuration items, see [TSO configuration file](/tso-configuration-file.md).
+- The instance-level `"-host"` configuration in `scheduling_servers` only supports IP address, not domain name.
```
Comment (Oreoxmt) [inline pd-microservices-deployment-topology.md:86]:
```suggestion
- The instance-level `host` configuration in `tso_servers` only supports IP address, not domain name.
- For detailed descriptions of TSO configuration items, see [TSO configuration file](/tso-configuration-file.md).
- The instance-level `host` configuration in `scheduling_servers` only supports IP address, not domain name.
```

File: scale-microservices-using-tiup.md [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+        port: 3379
+    ```
+
+2. Modify the cluster configuration and switch the cluster to PD microservice mode:
+
+    ```shell
+    tiup cluster edit-config <cluster-name>
+    ```
+
+   Add `pd_mode: ms` to `global`:
```
Comment (Oreoxmt) [inline scale-microservices-using-tiup.md:233]:
```suggestion
    Add `pd_mode: ms` to `global`:
```

File: scale-microservices-using-tiup.md [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+    global:
+    user: tidb
+    ssh_port: 22
+    listen_host: 0.0.0.0
+    deploy_dir: /tidb-deploy
+    data_dir: /tidb-data
+    os: linux
+    arch: amd64
+    systemd_mode: system
+    pd_mode: ms
```
Comment (qiancai) [inline scale-microservices-using-tiup.md:245]:
```suggestion
    global:
      user: tidb
       ssh_port: 22
       listen_host: 0.0.0.0
       deploy_dir: /tidb-deploy
       data_dir: /tidb-data
       os: linux
       arch: amd64
       systemd_mode: system
       pd_mode: ms
```

File: scale-microservices-using-tiup.md [hunk 10]
```diff
@@ -0,0 +1,302 @@
...
+    ```ini
+    global:
+    user: tidb
+    ssh_port: 22
+    listen_host: 0.0.0.0
+    deploy_dir: /tidb-deploy
+    data_dir: /tidb-data
+    os: linux
+    arch: amd64
+    systemd_mode: system
```
Comment (qiancai) [inline scale-microservices-using-tiup.md:285]:
```suggestion
    global:
      user: tidb
       ssh_port: 22
       listen_host: 0.0.0.0
       deploy_dir: /tidb-deploy
       data_dir: /tidb-data
       os: linux
       arch: amd64
       systemd_mode: system
```

**Review by rleungx — APPROVED** [review]

File: command-line-flags-for-scheduling-configuration.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+- Default: `http://127.0.0.1:2379`
+
+## `--cacert`
+
+- The file path of CA, used to enable TLS.
+- Default: `""`
+
+## `--cert`
+
+- The path of the PEM file including the X509 certificate, used to enable TLS.
```
Comment (Oreoxmt) [inline command-line-flags-for-scheduling-configuration.md:29]:
```suggestion
- The path of the PEM file including the X.509 certificate, used to enable TLS.
```

File: command-line-flags-for-scheduling-configuration.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+- If you specify a configuration file, the Scheduling node first reads configurations from that file. If the same configurations are also specified via command line flags, the Scheduling node uses the command line flag configurations to overwrite those in the configuration file.
+
+## `--data-dir`
+
+- The path to the data directory on the Scheduling node.
+- Default: `"default.${name}"`
+
+## `--key`
+
+- The path of the PEM file including the X509 key, used to enable TLS.
```
Comment (Oreoxmt) [inline command-line-flags-for-scheduling-configuration.md:45]:
```suggestion
- The path of the PEM file including the X.509 key, used to enable TLS.
```

File: command-line-flags-for-tso-configuration.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+- Default: `http://127.0.0.1:2379`
+
+## `--cacert`
+
+- The file path of CA, used to enable TLS.
+- Default: `""`
+
+## `--cert`
+
+- The path of the PEM file including the X509 certificate, used to enable TLS.
```
Comment (Oreoxmt) [inline command-line-flags-for-tso-configuration.md:29]:
```suggestion
- The path of the PEM file including the X.509 certificate, used to enable TLS.
```

File: command-line-flags-for-tso-configuration.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+- If you specify a configuration file, the TSO node first reads configurations from that file. If the same configurations are also specified via command line flags, the TSO node uses the command line flag configurations to overwrite those in the configuration file.
+
+## `--data-dir`
+
+- The path to the data directory on the TSO node.
+- Default: `"default.${name}"`
+
+## `--key`
+
+- The path of the PEM file including the X509 key, used to enable TLS.
```
Comment (Oreoxmt) [inline command-line-flags-for-tso-configuration.md:45]:
```suggestion
- The path of the PEM file including the X.509 key, used to enable TLS.
```

File: scheduling-configuration-file.md [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+Configuration items related to security
+
+### `cacert-path`
+
+- The path of the CA file
+- Default value: ""
+
+### `cert-path`
+
+- The path of the Privacy Enhanced Mail (PEM) file that contains the X509 certificate
```
Comment (Oreoxmt) [inline scheduling-configuration-file.md:62]:
```suggestion
- The path of the Privacy Enhanced Mail (PEM) file that contains the X.509 certificate
```

File: scheduling-configuration-file.md [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+- Default value: ""
+
+### `key-path`
+
+- The path of the PEM file that contains the X509 key
```
Comment (Oreoxmt) [inline scheduling-configuration-file.md:67]:
```suggestion
- The path of the PEM file that contains the X.509 key
```

File: tso-configuration-file.md [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+Configuration items related to security
+
+### `cacert-path`
+
+- The path of the CA file
+- Default value: ""
+
+### `cert-path`
+
+- The path of the Privacy Enhanced Mail (PEM) file that contains the X509 certificate
```
Comment (Oreoxmt) [inline tso-configuration-file.md:70]:
```suggestion
- The path of the Privacy Enhanced Mail (PEM) file that contains the X.509 certificate
```

File: tso-configuration-file.md [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+- Default value: ""
+
+### `key-path`
+
+- The path of the PEM file that contains the X509 key
```
Comment (Oreoxmt) [inline tso-configuration-file.md:75]:
```suggestion
- The path of the PEM file that contains the X.509 key
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**